### DISCUSSION TITLE
NestJS

### DISCUSSION THREAD

**Discussion by CodeRider101** [SE-TINF22B2/G2-Schoolify#60] [Technologien]
Warum benutzen wir Nest.js und nicht andere Technologien auf dieser Ebene?
@Schmarvinius @CodeRider101

**Comment by CodeRider101** [comment]
Nest.js ist eines der am schnellsten wachsenden [Node.js-Frameworks](https://kinsta.com/de/wissensdatenbank/was-ist-node-js/) für den Aufbau effizienter, skalierbarer und unternehmenstauglicher Backend-Anwendungen mit Node.js. Es ist dafür bekannt, dass es mit modernem JavaScript und TypeScript hochgradig testbare, wartbare und skalierbare Anwendungen erstellt. ([Quelle](https://kinsta.com/de/wissensdatenbank/nestjs/)).

Zusätzlich bietet es eine sehr gute Architektur für schnelle Entwicklung direkt zu Beginn. Sicherheitsaspekte sind vom Framework mit inbegriffen.

Nest.js ist einfach zu erlernen und zu beherrschen und verfügt über eine leistungsstarke CLI (Kommandozeile), die die Produktivität und die Entwicklung erleichtert. Mit dieser leistungsstarken CLI ist es ein Kinderspiel, jedes serverseitige Projekt zu starten und zu Ende zu bringen.
Darüber hinaus verfügt Nest.js über eine ausführliche Dokumentation und eine sehr aktive Gemeinschaft von Entwicklern und Mitwirkenden, die bereit sind, auf Probleme zu reagieren.

Ein Nachteil ist der Debugger. Den muss man mühsam aufsetzen und sich zuerst reinfuchsen

--> Trotzdem überwiegen die positiven Aspekte und deshalb haben wir uns für Nest.js entschieden

**Comment by Schmarvinius** [comment]
Wir haben uns für NestJS als unser BackEnd Framework entschieden. 
Da unser Schulplanner ein sehr komplexes Projekt wird, ist uns ein modulares und skalierbares Framework wichtig. 
NestJS steht genau für Modularität und Skalierbarkeit, es macht uns möglich unser Projekt in viele kleinere, unabhängige Module zu unterteilen, dies erleichtert Wartung und Skalierung.
Dazu bietet NestJS uns eine Reihe von Funktionen und Werkzeugen an, die uns die Entwicklung erleichtert. Dazu gehören unter anderem generisches API-Framework, ein Dependency Injection-Framework und ein Testframework.
Sicherheit der Daten liegt uns natürlich auch am Herzen, NestJS bietet auch dafür eingebaute Funktionen wie zum Beispiel Authentifizierung, Autorisierung, TLS-Verschlüsselung.


NestJS besitzt ebenfalls eine herausragende Dokumentation, die für alle Funktionen des Frameworks eine Erklärung wie auch meist ein Code Beispiel besitzt.
Dazu bietet NestJS eine CLI die den Entwickler bei der Initialisierung, Entwicklung und Wartung unterstützt. Die NestJS CLI verwendet best-practice Architektur Schemata um gut strukturierte Applikationen zu fördern.

Nachteile dieses Frameworks für unser Projekt ist einerseits seine steile Lernkurve und die fehlende Erfahrung mit diesem Framework in unserem Team
